src/app/components/blog/blog.component.ts

```diff
@@ -64,7 +84,33 @@ export class BlogComponent implements OnInit {
     return null;
   }
 
-  addNewPost(): void {}
+  addNewPost(): void {
+  }
 
-  editPost(id: string): void {}
+  editPost(id: string): void {
+    const postToEdit = this.blogPosts().find((post) => post.id === id);
```

Comment:
How are you editing a post here?  you simply find the post by its ID and open the modal.